### PR TITLE
Move questions search functionality to questions_search controller

### PR DESCRIPTION
**Pull request by ananyo2012**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] tests pass -- `rake test`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request are descriptively named
* [x] if possible, multiple commits squashed if they're smaller changes
* [ ] reviewed/confirmed/tested by another contributor or maintainer
* [x] `development.sqlite.example` has been updated if any database migrations were added

As Promised all the search functionalities for questions are moved to a new questions_search controller. Corresponding changes are made to views and tests.

Thanks!

### CODE REVIEW

**Comment by ananyo2012** [comment]
@jywarren Please review the changes made. I have also covered the points mentioned in #554 for search page and autocomplete results. The autocomplete results look like:
questions_search

File: app/controllers/questions_search_controller.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                  .joins(:drupal_tag)
+                  .where('term_data.name LIKE ?', 'question:%')
+                  .order('node.nid DESC')
+                  .limit(25)
+    questions.each do |match|
+      link = "<span class='pull-left'><i data-url='" + match.path(:question) + 
+             "' class='fa fa-question-circle'></i> " + match.title + "</span>"
+      author = "<span class='pull-right'><i class='fa fa-user'></i> " + match.author.name
+      likes = " | <i class='fa fa-star'></i> " + match.cached_likes.to_s + "</span>"
+      # answer = " | <i class='fa f-comments'></i> " + match.answers.length.to_s
```
Comment (ananyo2012) [inline app/controllers/questions_search_controller.rb]: Commented out for the time being. Can be implemented after #566 is merged

**Comment by ananyo2012** [comment]
@ujithaperera @david-days You all can rest easy now. I have moved the questions based search functionality to a  new questions_search controller.

**Comment by ujithaperera** [comment]
@ananyo2012,
Yes thanks to @David, we both can work parallely. And let us know when you finish with this controller. Then we csn merge this controller to our fork and do necessary updates. 
Cheers.

**Comment by ananyo2012** [comment]
@ujithaperera You mean @david-days :stuck_out_tongue: Please edit so that someone else won't be notified.

**Comment by jywarren** [comment]
Looking great, folks, thank you for your excellent coordination. Ananyo, can you add a brief comment describing that the separate controller is temporary, to the top of the controller?

Once I've merged #566 tomorrow morning, can i merge this on top, or will it need the commented out section re-added?

**Comment by ananyo2012** [comment]
Sure Jeff adding comments now. Yes you can add merge this on the top. Yes do need to uncomment the answer section to see the answers on the autocomplete results. Also move the `</span>`, the closing span tag from likes to answers. Or if you insists I can make the changes now. But you may see some tests failing as it will need the answers feature merged.

**Comment by jywarren** [comment]
I rebase each merge in order of time, and then test, so if the tests will
pass after the rebase, that's all good!

**Comment by ananyo2012** [comment]
@jywarren Added Comments and edits in code.

**Comment by jywarren** [comment]
perfect. Thanks, Ananyo - you've put in a huge amount of work today and I
appreciate it!

**Comment by jywarren** [comment]
Rebasing this was a bit tough -- a couple conflicts, probably due to the very large # of PRs. Running tests now.

**Comment by jywarren** [comment]
Manually merged!

**Comment by ananyo2012** [comment]
@jywarren Thanks for your  patience Jeff! This PR was important for both Advanced Search Project and Q & A Project.

**Comment by jywarren** [comment]
:-)

**Pull request closed without merging**